internal/chezmoi/sourcestate.go

```diff
@@ -2338,6 +2345,17 @@ func (s *SourceState) readExternalArchive(
 			if err != nil {
 				return fmt.Errorf("%s: %w", name, err)
 			}
+
+			if strings.HasPrefix(path.Base(name), "._") && !external.Archive.ExtractAppleDoubleFiles {
+				if bytes.Equal(appleDoubleMagicCode, contents[0:4]) {
+					return nil
+				}
+
+				if bytes.Equal(appleDoubleVersion, contents[4:8]) {
+					return nil
+				}
+			}
+
```

Comment:
These lines are identical to the lines in `SourceState.readExternalArchiveFile`, so I'd factor them out into a `isAppleDoubleFile(name string, contents []byte) bool` function.